Feature/pelagos 3523 basic search form and button

praneethpr opened the pull request.

fightingtexasaggie commented on src/Pelagos/Bundle/AppBundle/Resources/views/Search/results.html.twig:
```diff
@@ -0,0 +1,60 @@
+                                    {{ result.year |default }}
+                                    {{ result.title }}
+                                    {{ result.researchGroup.name }}.
+                                    DOI: {{ result.doi.doi }}
```
The doi might possibly be null, so a |default could help.

praneethpr commented:
> You may want to look at the util/DatasetIndex class starting at line 80. This is how we handled the search by UDI and search by DOI, otherwise any UDI returns a ton of unrelated datasets. By default the UDI is split into 4 parts and each is search for separately. We regexp matched an UDI and a DOI, and made those exact-phrase only.

Udi search wasn't a part of this ticket, i just added that for testing. Removed it
